### PR TITLE
DIV-4630 - added RespSolLinkedEmail field to flag linked sol cases

### PR DESCRIPTION
**Pull request by hosainnet**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines

### JIRA link (if applicable) ###

DIV-4630

### Change description ###

* Added RespSolLinkedEmail field to set the field, which solicitor can read/write
* courtadmin can read the field so that our system user can read this field, to check whether a case is already linked or not before proceeding - see https://github.com/hmcts/div-case-orchestration-service/pull/337
* Added permission for solicitor be able to update the case in the AwaitingAosSol state, so that they can set the field above on the case once they link themselfs 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 1]
```diff
@@ -12361,6 +12361,20 @@
     "UserRole": "caseworker-divorce-courtadmin",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "26/06/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "RespSolLinkedEmail",
+    "UserRole": "caseworker-divorce-solicitor",
+    "CRUD": "CRU"
+  },
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: Solicitor role can update

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 2]
```diff
@@ -12361,6 +12361,20 @@
...
+  {
+    "LiveFrom": "26/06/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "RespSolLinkedEmail",
+    "UserRole": "caseworker-divorce-courtadmin",
+    "CRUD": "R"
+  },
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: Service user can read

File: definitions/divorce/json/AuthorisationCaseState.json [hunk 3]
```diff
@@ -1264,7 +1264,7 @@
     "CaseTypeID": "DIVORCE",
     "CaseStateID": "AosAwaitingSol",
     "UserRole": "caseworker-divorce-solicitor",
-    "CRUD": "R"
+    "CRUD": "CRU"
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseState.json]: So that solicitor can set the field

File: definitions/divorce/json/CaseField.json [hunk 4]
```diff
@@ -2201,6 +2201,14 @@
...
     "SecurityClassification": "Public"
   },
+  {
+    "LiveFrom": "26/06/2019",
+    "CaseTypeID": "DIVORCE",
+    "ID": "RespSolLinkedEmail",
+    "Label": "Respondent solicitor's email address they used to link the case",
+    "FieldType": "Email",
+    "SecurityClassification": "Public"
+  },
```
Comment (hosainnet) [inline definitions/divorce/json/CaseField.json]: The field

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 2 above)
Comment (tchow8) [reply definitions/divorce/json/AuthorisationCaseField.json]: Does beta user already have READ? I believe now caseworkers either have courtadmin or courtadmin_beta role, but not both right?

Comment (hosainnet) [reply definitions/divorce/json/AuthorisationCaseField.json]: I really only need the service token to read this, not a human/caseworker - is systemupdate better for that?

**Pull request merged**